tests/ignite/metrics/nlp/test_bleu.py

```diff
@@ -121,23 +121,52 @@ def test_corpus_bleu_smooth2(candidate, references):
             )
         bleu = Bleu(ngram=i, smooth="smooth2")
         assert reference == bleu._corpus_bleu(references, candidate)
-        bleu.update((candidate[0], references[0]))
+        bleu.update((candidate, references))
         assert reference == bleu.compute()
 
 
 def test_bleu():
     bleu = Bleu(ngram=4, smooth="smooth2")
-    bleu.update((corpus.cand_1, corpus.references_1))
-    bleu.update((corpus.cand_2a, corpus.references_2))
-    bleu.update((corpus.cand_2b, corpus.references_2))
-    bleu.update((corpus.cand_3, corpus.references_2))
+    bleu.update(([corpus.cand_1], [corpus.references_1]))
+    bleu.update(([corpus.cand_2a], [corpus.references_2]))
+    bleu.update(([corpus.cand_2b], [corpus.references_2]))
+    bleu.update(([corpus.cand_3], [corpus.references_2]))
     value = bleu._corpus_bleu([corpus.references_1], [corpus.cand_1])
     value += bleu._corpus_bleu([corpus.references_2], [corpus.cand_2a])
     value += bleu._corpus_bleu([corpus.references_2], [corpus.cand_2b])
     value += bleu._corpus_bleu([corpus.references_2], [corpus.cand_3])
     assert bleu.compute() == value / 4
 
 
+def test_bleu_batch():
+
+    # Batch size 1
+    refs = [corpus.references_2]
+    hypotheses = [corpus.cand_2a]
+    bleu = Bleu(ngram=4)
+    bleu.update((hypotheses, refs))
+
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore")
+        reference_bleu_score = (corpus_bleu([refs[0]], [hypotheses[0]]) + corpus_bleu([refs[0]], [hypotheses[0]])) / 2
+    assert bleu.compute() == reference_bleu_score
+
```

Comment:
It would be nice to compare the batch version, batch 1 version (loop over the batch) and the nltk bleu as you did.

Reply:
Added [here](https://github.com/pytorch/ignite/pull/2180/files#diff-4f64ad98d488d32957c8f58462fdb23ab0c3ea93f7b8150ff49c2a39665074b4R159-R170). This is what you were looking for right?